src/releaseProvider.ts

```diff
@@ -0,0 +1,77 @@
+import { env, ExtensionContext } from "vscode";
+import * as vscode from "vscode";
+import * as semver from "semver";
+
+const versionKey = "metals-server-version";
+
+export class ReleaseProvider {
+  private readonly state: typeof this.context.globalState;
+
+  constructor(
+    private readonly context: ExtensionContext,
+    // TODO switch to extension version maybe?
+    private readonly currentServerVersion: string
+  ) {
+    this.state = context.globalState;
+  }
+
+  showWhatsNewIfNeeded() {
+    if (this.shouldShowPanel) {
+      this.showPanel();
+    }
+  }
+
+  private showPanel() {
+    const previousValueOpt: string | undefined = this.state.get(versionKey);
+
+    const previousValue = previousValueOpt || this.currentServerVersion;
+    const isNewerVersion = this.isNewerVersion(previousValue);
+
+    if (isNewerVersion) {
+      const panel = vscode.window.createWebviewPanel(
+        `scalameta.metals.whatsNew`,
+        `What's New in Metals`,
+        vscode.ViewColumn.One
+      );
+
+      // TODO better content
```

Comment:
Maybe we could just embed the full release notes? We could create aliases like `https://scalameta.org/metals/0.11.6` etc.

Reply:
You could also get the `body` property from `https://api.github.com/repos/scalameta/metals/releases/tags/v0.11.6`

This wouldn't require a manual step with the alias (unless we can create them automatically? :thinking: )

Reply:
I did the request to the `https://scalameta.org/metals/blog/2022/06/03/aluminium` similar to [this](https://github.com/scalameta/metals-vscode/blob/75fbfe51fbe248c4c13fef9ba8d66201dd10f24b/src/getServerVersion.ts#L75-L85) but I got "301 Moved Permanently". Not sure why though

Reply:
Fun fact, if you use `https://scalameta.org/metals/blog/2022/06/03/aluminium/` it works correctly with the added `/`

Reply:
At least for now I stayed with markdown and parsing it to html - it was already done and I'm not very keen of cutting fragment of html from whole site. But I'm don't have strong opinion on that.

Reply:
Crazy idea:
- obtain `body` from `https://api.github.com/repos/scalameta/metals/releases/tags/`
- extract `2022/06/03/aluminium` part
- switch it to `2022-06-03-aluminium`
- make request to the `https://raw.githubusercontent.com/scalameta/metals/main/website/blog/2022-06-03-aluminium.md`

Doing request to the `https://api.github.com/repos/scalameta/metals/releases/tags/v0.11.6` needs having user-agent set. I'll explore that possibility tommorow.

Reply:
`getMarkdownLink` retrieves link to markdown file with release notes "automagically", there no need to specify it manually! Code is very ugly, but work, I'll refactor this later if approach is ok. @tgodzik